test/envtest/sgxdeviceplugin_controller_test.go

```diff
@@ -59,16 +62,66 @@ var _ = Describe("SgxDevicePlugin Controller", func() {
 				return len(fetched.Status.ControlledDaemonSet.UID) > 0
 			}, timeout, interval).Should(BeTrue())
 
-			By("updating image name successfully")
+			By("checking DaemonSet is created successfully")
+			ds := &apps.DaemonSet{}
+			_ = k8sClient.Get(context.Background(), types.NamespacedName{Namespace: ns, Name: "intel-sgx-plugin"}, ds)
+			Expect(ds.Spec.Template.Spec.NodeSelector).To(Equal(fetched.Spec.NodeSelector))
+			Expect(ds.Spec.Template.Spec.InitContainers).To(HaveLen(1))
+			Expect(ds.Spec.Template.Spec.InitContainers[0].Image).To(Equal(spec.InitImage))
+
+			By("updating SgxDevicePlugin successfully")
 			updatedImage := "updated-sgx-testimage"
+			updatedInitImage := "updated-testinitimage"
+			updatedLogLevel := 2
+			updatedEnclaveLimit := 2
+			updatedProvisionLimit := 2
+			updatedNodeSelector := map[string]string{"test": "updated-nodeselector"}
+
 			fetched.Spec.Image = updatedImage
+			fetched.Spec.InitImage = updatedInitImage
+			fetched.Spec.LogLevel = updatedLogLevel
+			fetched.Spec.EnclaveLimit = updatedEnclaveLimit
+			fetched.Spec.ProvisionLimit = updatedProvisionLimit
+			fetched.Spec.NodeSelector = updatedNodeSelector
 
 			Expect(k8sClient.Update(context.Background(), fetched)).Should(Succeed())
 			fetchedUpdated := &devicepluginv1.SgxDevicePlugin{}
 			Eventually(func() string {
 				_ = k8sClient.Get(context.Background(), key, fetchedUpdated)
 				return fetchedUpdated.Spec.Image
 			}, timeout, interval).Should(Equal(updatedImage))
```

Comment:
one thing to note about these lines is that now we change many other fields too but don't change the assertion

Reply:
@mythi
Yeah, I think so too! If device plugin is not set properly, daemonset cannot also have the same value.
So, we don't need to add anything more, right? (Did I understand correctly?)

Reply:
the question becomes do we want to drop that CR `Spec.Image` check

Reply:
Ah, true.
I think it makes sense to check it also with others while checking if daemonset is created successfully or not.